### PR TITLE
Replace deprecated NewDeltaFIFO with NewDeltaFIFOWithOptions

### PR DESCRIPTION
**Pull request by onesafe**
What type of PR is this?
/kind cleanup

What this PR does / why we need it:
In some test file, we still use deprecated func NewDeltaFIFO. I have use NewDeltaFIFOWithOptions replace the deprecated func in test file.

Does this PR introduce a user-facing change?
```release-note
None
```

### CODE REVIEW

**Comment by onesafe** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

kind: test

File: staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go [hunk 1]
```diff
@@ -56,7 +56,10 @@ func (kl literalListerGetter) GetByKey(key string) (interface{}, bool, error) {
 }
 
 func TestDeltaFIFO_basic(t *testing.T) {
-	f := NewDeltaFIFO(testFifoObjectKeyFunc, nil)
+	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
+		KeyFunction: testFifoObjectKeyFunc,
+		KnownObjects: nil,
```
Comment (ingvagabund) [inline staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:61]: `KnownObjects` is nil by default if not set

Comment (onesafe) [reply staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:61]: Got it, i will update this pr, thank you.

**Comment by ingvagabund** [comment]
```
$ grep -rn "NewDeltaFIFO" --include "*.go"
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:59:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: testFifoObjectKeyFunc})
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:103:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:124:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: testFifoObjectKeyFunc})
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:168:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: testFifoObjectKeyFunc})
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:206:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: testFifoObjectKeyFunc})
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:227:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:251:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: testFifoObjectKeyFunc})
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:277:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:296:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:314:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:337:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:362:	f = NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:388:	f = NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: testFifoObjectKeyFunc})
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:409:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:439:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:461:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:483:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:514:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: testFifoObjectKeyFunc})
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:542:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: testFifoObjectKeyFunc})
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:648:		f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: testFifoObjectKeyFunc})
staging/src/k8s.io/client-go/tools/cache/delta_fifo_test.go:662:	f := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/controller_test.go:47:	fifo := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/shared_informer.go:371:	fifo := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/controller.go:381:	fifo := NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:166:// NewDeltaFIFO returns a Queue which can be used to process changes to items.
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:203:// Use `NewDeltaFIFOWithOptions(DeltaFIFOOptions{..., EmitDeltaTypeReplaced: true})`
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:206:// Deprecated: Equivalent to NewDeltaFIFOWithOptions(DeltaFIFOOptions{KeyFunction: keyFunc, KnownObjects: knownObjects})
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:207:func NewDeltaFIFO(keyFunc KeyFunc, knownObjects KeyListerGetter) *DeltaFIFO {
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:208:	return NewDeltaFIFOWithOptions(DeltaFIFOOptions{
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:214:// NewDeltaFIFOWithOptions returns a Queue which can be used to process changes to
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:216:func NewDeltaFIFOWithOptions(opts DeltaFIFOOptions) *DeltaFIFO {
```
looks clean

**Comment by ingvagabund** [comment]
Deprecated through https://github.com/kubernetes/kubernetes/pull/86015 (Dec 2019, 1.18). We can drop the deprecated `NewDeltaFIFO` in 1.22 (after 3 releases).

**Comment by ingvagabund** [comment]
@onesafe this PR will be here for a while until 1.21 is out and the code freeze lifted though

**Comment by onesafe** [comment]
> @onesafe this PR will be here for a while until 1.21 is out and the code freeze lifted though

understand, thank you.

**Pull request merged**